Calibration/EcalCalibAlgos/src/EcalPhiSymRecHit.cc

```diff
@@ -0,0 +1,60 @@
+#include "Calibration/EcalCalibAlgos/interface/EcalPhiSymRecHit.h"
+
+//**********constructors******************************************************************
+EcalPhiSymRecHit::EcalPhiSymRecHit()
+    : id_(0), chStatus_(0), nHits_(0), etSum_(1, 0.), et2Sum_(0), lcSum_(0), lc2Sum_(0) {}
+
+EcalPhiSymRecHit::EcalPhiSymRecHit(uint32_t id, unsigned int nMisCalibV, unsigned int status)
+    : id_(id), chStatus_(status), nHits_(0), etSum_(nMisCalibV, 0.), et2Sum_(0), lcSum_(0), lc2Sum_(0) {}
+
+EcalPhiSymRecHit::EcalPhiSymRecHit(uint32_t id, std::vector<float>& etValues, unsigned int status)
+    : id_(id),
+      chStatus_(status),
+      nHits_(0),
+      etSum_(etValues.begin(), etValues.end()),
+      et2Sum_(0),
+      lcSum_(0),
+      lc2Sum_(0) {}
+
+//**********destructor********************************************************************
+EcalPhiSymRecHit::~EcalPhiSymRecHit() {}
+
+//**********utils*************************************************************************
+
+void EcalPhiSymRecHit::AddHit(float* etValues, float laserCorr) {
+  if (etValues[0] > 0.) {
+    ++nHits_;
+    etSum_[0] += etValues[0];
+    et2Sum_ += etValues[0] * etValues[0];
+    lcSum_ += laserCorr;
+    lc2Sum_ += laserCorr * laserCorr;
+  }
+  for (unsigned int i = 0; i < etSum_.size(); ++i)
+    etSum_[i] += etValues[i];
```

Comment:
Is `etValues` guaranteed to be the same size than `etSum_`? Otherwise maybe
```suggestion
    etSum_[i] += etValues.at(i);
```

Reply:
This would not work here because etValues is a bare pointer. I have anyway opted to remove this method and leave only the one in which etValues is a vector. I implemented a proper check.